test/e2e/specs/site-editor/style-variations.spec.js

```diff
@@ -0,0 +1,299 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.use( {
+	siteEditorStyleVariations: async (
+		{ admin, editor, page, pageUtils, requestUtils },
+		use
+	) => {
+		await use(
+			new SiteEditorStyleVariations( {
+				admin,
+				editor,
+				page,
+				pageUtils,
+				requestUtils,
+			} )
+		);
+	},
+} );
+
+test.describe( 'Global styles variations', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme(
+			'gutenberg-test-themes/style-variations'
+		);
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await Promise.all( [
+			requestUtils.deleteAllTemplates( 'wp_template' ),
+			requestUtils.deleteAllTemplates( 'wp_template_part' ),
+		] );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test( 'should have three variations available with the first one being active', async ( {
+		admin,
+		editor,
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await editor.toggleGlobalStyles();
+		await siteEditorStyleVariations.openOtherStyles();
+
+		const variations = await siteEditorStyleVariations.getAvailableStyleVariations();
+
+		expect( await variations.count() ).toEqual( 3 );
+		expect( await variations.first().getAttribute( 'class' ) ).toContain(
+			'is-active'
+		);
+		expect(
+			await variations.nth( 1 ).getAttribute( 'class' )
+		).not.toContain( 'is-active' );
+		expect(
+			await variations.nth( 2 ).getAttribute( 'class' )
+		).not.toContain( 'is-active' );
+	} );
+
+	test( 'should apply preset colors and font sizes in a variation', async ( {
+		admin,
+		editor,
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await siteEditorStyleVariations.applyPinkVariation();
+		await editor.openPreviousGlobalStylesPanel();
+
+		await siteEditorStyleVariations.openColorsPanel();
+		expect(
+			await siteEditorStyleVariations.getBackgroundColorValue()
+		).toBe( 'rgb(202, 105, 211)' );
```

Comment:
I wonder if we could instead use [`.toHaveCSS()`](https://playwright.dev/docs/test-assertions#locator-assertions-to-have-css) here?

Reply:
Ooh, nice! Thanks for the tip.